<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Calculation of Haplotype Trait Relative â€¢ HaploTraitR</title>
<script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet">
<link href="deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet">
<script src="deps/headroom-0.11.0/headroom.min.js"></script><script src="deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="deps/search-1.0.0/fuse.min.js"></script><script src="deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="pkgdown.js"></script><meta property="og:title" content="Calculation of Haplotype Trait Relative">
<meta name="description" content="This project performs haplotype analysis and visualizes phenotypic data across multiple genetic/haplotype combinations to understand the association between haplotypes (SNPs linked by LD) and variation in traits.">
<meta property="og:description" content="This project performs haplotype analysis and visualizes phenotypic data across multiple genetic/haplotype combinations to understand the association between haplotypes (SNPs linked by LD) and variation in traits.">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="index.html">HaploTraitR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="reference/index.html">Reference</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="search.json">
</form></li>
      </ul>
</div>


  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><p><link rel="stylesheet" type="text/css" href="styles.css"></p>
<!-- Include the JavaScript file -->
<script src="team.js"></script><div class="section level1">
<div class="page-header"><h1 id="haplotype-analysis-project">Haplotype Analysis Project<a class="anchor" aria-label="anchor" href="#haplotype-analysis-project"></a>
</h1></div>
<p>This project performs haplotype analysis and visualizes phenotypic data across multiple genetic/haplotype combinations to understand the association between haplotypes (SNPs linked by LD) and variation in traits.</p>
<div class="section level2">
<h2 id="dependencies">Dependencies<a class="anchor" aria-label="anchor" href="#dependencies"></a>
</h2>
<p>Please install the following R packages before installing the <code>HaploTraitR</code> package:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Install required packages if not already installed</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"devtools"</span>, <span class="st">"Rcpp"</span>, <span class="st">"igraph"</span>, <span class="st">"ggplot2"</span>, <span class="st">"rstatix"</span>, <span class="st">"dplyr"</span>, <span class="st">"ggpubr"</span>, <span class="st">"reshape2"</span>, <span class="st">"gridExtra"</span>, <span class="st">"ggtext"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Install the HaploTraitR package from GitHub</span></span>
<span><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"AlsammanAlsamman/HaploTraitR"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="sample-files">Sample Files<a class="anchor" aria-label="anchor" href="#sample-files"></a>
</h2>
<ul>
<li>
<strong>Haplotype File:</strong> <code>Barley_50K_KNNimp.hmp.txt</code> - Contains SNP genotype information.</li>
<li>
<strong>GWAS File:</strong> <code>SignificantSNP_GWAS.csv</code> - Contains significant SNPs identified from a GWAS study.</li>
<li>
<strong>Phenotype File:</strong> <code>Pheno_ANN19.tsv</code> - Contains phenotypic data for the traits of interest.</li>
</ul>
</div>
<div class="section level2">
<h2 id="test-parameters">Test Parameters<a class="anchor" aria-label="anchor" href="#test-parameters"></a>
</h2>
<ul>
<li>
<strong>Distance Threshold:</strong> 1,000,000 (for clustering all SNPs within 1Mb of the significant SNPs)</li>
<li>
<strong>Cluster Count Threshold:</strong> 5 (minimum number of SNPs in a cluster)</li>
<li>
<strong>LD Threshold:</strong> 0.3 (minimum LD value to consider two SNPs in LD)</li>
<li>
<strong>Combination Frequency Threshold:</strong> 0.1 (minimum genotypes with the same combination to consider the combination)</li>
</ul>
</div>
<div class="section level2">
<h2 id="configuration-settings">Configuration Settings<a class="anchor" aria-label="anchor" href="#configuration-settings"></a>
</h2>
<p>The <code>HaploTraitR</code> package allows users to customize various configuration settings. Below are the default settings that can be changed by the user:</p>
<ul>
<li>
<strong>outfolder:</strong> <code>NULL</code> (Output folder for results)</li>
<li>
<strong>dist_threshold:</strong> <code>1000000</code> (Distance threshold for clustering SNPs within 1Mb)</li>
<li>
<strong>dist_cluster_count:</strong> <code>5</code> (Minimum number of SNPs in a cluster)</li>
<li>
<strong>ld_threshold:</strong> <code>0.3</code> (Minimum LD value to consider two SNPs in LD)</li>
<li>
<strong>comb_freq_threshold:</strong> <code>0.1</code> (Minimum frequency of genotypes with the same combination)</li>
<li>
<strong>fdr_threshold:</strong> <code>0.05</code> (False discovery rate threshold for multiple testing correction)</li>
<li>
<strong>fdr_method:</strong> <code>"fdr"</code> (Method for FDR correction)</li>
<li>
<strong>t_test_threshold:</strong> <code>0.05</code> (Threshold for t-test significance)</li>
<li>
<strong>t_test_method:</strong> <code>"t.test"</code> (Method for t-test)</li>
<li>
<strong>rsid_col:</strong> <code>"rsid"</code> (Column name for rsid in GWAS file)</li>
<li>
<strong>pos_col:</strong> <code>"pos"</code> (Column name for position in GWAS file)</li>
<li>
<strong>chr_col:</strong> <code>"chr"</code> (Column name for chromosome in GWAS file)</li>
<li>
<strong>pval_col:</strong> <code>"p"</code> (Column name for p-value in GWAS file)</li>
<li>
<strong>fdr_col:</strong> <code>"fdr"</code> (Column name for FDR in GWAS file)</li>
<li>
<strong>phenotypename:</strong> <code>"Phenotype"</code> (Name of the phenotype)</li>
<li>
<strong>phenotypeunit:</strong> <code>NULL</code> (Unit of the phenotype)</li>
</ul>
<p>To initialize the default configuration, use the <code>initialize_config</code> function:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">initialize_config</span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p>To set or get specific configuration values, use the <code>set_config</code> and <code>get_config</code> functions:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Set a configuration value</span></span>
<span><span class="fu"><a href="reference/set_config.html">set_config</a></span><span class="op">(</span>config <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>dist_threshold <span class="op">=</span> <span class="fl">500000</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Get a configuration value</span></span>
<span><span class="va">dist_threshold</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/get_config.html">get_config</a></span><span class="op">(</span><span class="st">"dist_threshold"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">dist_threshold</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="usage">Usage<a class="anchor" aria-label="anchor" href="#usage"></a>
</h2>
<div class="section level3">
<h3 id="example-workflow">Example Workflow<a class="anchor" aria-label="anchor" href="#example-workflow"></a>
</h3>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://alsammanalsamman.github.io/HaploTraitR/">HaploTraitR</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Step 1: Create a unique output folder for results</span></span>
<span><span class="va">result_folder</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/create_unique_result_folder.html">create_unique_result_folder</a></span><span class="op">(</span>location <span class="op">=</span> <span class="st">"sampleout"</span><span class="op">)</span></span>
<span><span class="fu"><a href="reference/set_config.html">set_config</a></span><span class="op">(</span>config <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>outfolder <span class="op">=</span> <span class="va">result_folder</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Step 2: Read Data Files</span></span>
<span><span class="va">gwasfile</span> <span class="op">&lt;-</span> <span class="st">"sampledata/SignificantSNP_GWAS.tsv"</span></span>
<span><span class="va">hapfile</span> <span class="op">&lt;-</span> <span class="st">"sampledata/Barley_50K_KNNimp.hmp.txt"</span></span>
<span><span class="va">phenofile</span> <span class="op">&lt;-</span> <span class="st">"sampledata/Pheno_ANN19.tsv"</span></span>
<span></span>
<span><span class="va">gwas</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/readGWAS.html">readGWAS</a></span><span class="op">(</span><span class="va">gwasfile</span>, sep <span class="op">=</span> <span class="st">"\t"</span><span class="op">)</span></span>
<span><span class="va">hapmap</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/readHapmap.html">readHapmap</a></span><span class="op">(</span><span class="va">hapfile</span><span class="op">)</span></span>
<span><span class="va">pheno</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.csv</a></span><span class="op">(</span><span class="va">phenofile</span>, header <span class="op">=</span> <span class="cn">TRUE</span>, sep <span class="op">=</span> <span class="st">"\t"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Step 3: Set phenotype information in the configuration</span></span>
<span><span class="fu"><a href="reference/set_config.html">set_config</a></span><span class="op">(</span>config <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>phenotypename <span class="op">=</span> <span class="st">"PH"</span>, phenotypeunit <span class="op">=</span> <span class="st">"cm"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Step 4: Generate Boxplots for Genotype-Phenotype Distribution</span></span>
<span><span class="va">bxplts</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/boxplot_genotype_phenotype.html">boxplot_genotype_phenotype</a></span><span class="op">(</span><span class="va">pheno</span>, <span class="va">gwas</span>, <span class="va">hapmap</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Step 5: Cluster SNPs by Chromosome</span></span>
<span><span class="va">dist_clusters</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/getDistClusters.html">getDistClusters</a></span><span class="op">(</span><span class="va">gwas</span>, <span class="va">hapmap</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Step 6: Compute LD Matrices</span></span>
<span><span class="va">LDsInfo</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/computeLDclusters.html">computeLDclusters</a></span><span class="op">(</span><span class="va">hapmap</span>, <span class="va">dist_clusters</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Step 7: Cluster LD Values and Convert to Haplotypes</span></span>
<span><span class="va">clusterLDs</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/getLDclusters.html">getLDclusters</a></span><span class="op">(</span><span class="va">LDsInfo</span><span class="op">)</span></span>
<span><span class="va">haplotypes</span> <span class="op">&lt;-</span> <span class="fu">HaploTraitR</span><span class="fu">::</span><span class="fu"><a href="reference/convertLDclusters2Haps.html">convertLDclusters2Haps</a></span><span class="op">(</span><span class="va">hapmap</span>, <span class="va">clusterLDs</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Step 8: Get Sample Haplotypes and Save</span></span>
<span><span class="va">snpcombsample</span> <span class="op">&lt;-</span> <span class="fu">HaploTraitR</span><span class="fu">::</span><span class="fu"><a href="reference/getHapCombSamples.html">getHapCombSamples</a></span><span class="op">(</span><span class="va">haplotypes</span>, <span class="va">hapmap</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Step 9: Generate SNP Combination Tables</span></span>
<span><span class="va">SNPcombTables</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/getSNPcombTables.html">getSNPcombTables</a></span><span class="op">(</span><span class="va">snpcombsample</span>, <span class="va">pheno</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Step 10: Perform t-tests on SNP Combinations</span></span>
<span><span class="va">t_test_snpComp</span> <span class="op">&lt;-</span> <span class="fu">HaploTraitR</span><span class="fu">::</span><span class="fu"><a href="reference/testSNPcombs.html">testSNPcombs</a></span><span class="op">(</span><span class="va">SNPcombTables</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Step 11: Generate and Save Haplotype Combination Boxplots</span></span>
<span><span class="fu"><a href="reference/generateHapCombBoxPlots.html">generateHapCombBoxPlots</a></span><span class="op">(</span><span class="va">SNPcombTables</span>, <span class="va">t_test_snpComp</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Step 12: Plot Haplotype Distribution</span></span>
<span><span class="va">lDplots</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/plotLDForClusters.html">plotLDForClusters</a></span><span class="op">(</span><span class="va">clusterLDs</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Step 13: Plot LD Combination Matrix with LD</span></span>
<span><span class="va">LDCombs</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/plotLDCombMatrix.html">plotLDCombMatrix</a></span><span class="op">(</span><span class="va">clusterLDs</span>, <span class="va">haplotypes</span>, <span class="va">gwas</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="example-output">Example Output<a class="anchor" aria-label="anchor" href="#example-output"></a>
</h2>
<p>Here are some example plots generated using the <code>HaploTraitR</code> package:</p>
<div class="section level3">
<h3 id="haplotype-combination-boxplot">Haplotype Combination Boxplot<a class="anchor" aria-label="anchor" href="#haplotype-combination-boxplot"></a>
</h3>
<div class="float">
<img src="reference/figures/7H:501592821_boxplot.png" alt="Haplotype Combination Boxplot"><div class="figcaption">Haplotype Combination Boxplot</div>
</div>
</div>
<div class="section level3">
<h3 id="ld-combination-matrix">LD Combination Matrix<a class="anchor" aria-label="anchor" href="#ld-combination-matrix"></a>
</h3>
<div class="float">
<img src="reference/figures/7H:501592821_ld_comb_matrix.png" alt="LD Combination Matrix"><div class="figcaption">LD Combination Matrix</div>
</div>
</div>
</div>
<div class="section level2">
<h2 id="team-members">Team Members<a class="anchor" aria-label="anchor" href="#team-members"></a>
</h2>
</div>
<div class="section level2">
<h2 id="team">Team<a class="anchor" aria-label="anchor" href="#team"></a>
</h2>



<meta charset="UTF-8">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>
Team Section
</title>
<link rel="stylesheet" type="text/css" href="man/figures/styles.css">
<h2 id="team-1">
Team
<a class="anchor" aria-label="anchor" href="#team-1"></a>
</h2>
<div id="team-container" class="team-container">
<p><!-- Team members will be dynamically inserted here by JavaScript --></p>
</div>
<script>
document.addEventListener("DOMContentLoaded", function() {
  const teamContainer = document.getElementById("team-container");

  const teamMembers = [
    {
      name: "Dr. Zakaria Kehel",
      affiliation: "genetic resources scientist and senior biometrician",
      img: "man/figures/zk.jpg"
    },
    {
      name: "Alsamman Alsamman",
      affiliation: "ICARDA Bioinformatics consultant",
      img: "man/figures/ama.png"
    }
  ];

  teamMembers.forEach(member => {
    const memberDiv = document.createElement("div");
    memberDiv.className = "team-member";

    const img = document.createElement("img");
    img.src = member.img;
    img.alt = member.name;
    img.width = 150;
    img.height = 150;

    const name = document.createElement("h3");
    name.textContent = member.name;

    const affiliation = document.createElement("p");
    affiliation.textContent = member.affiliation;

    memberDiv.appendChild(img);
    memberDiv.appendChild(name);
    memberDiv.appendChild(affiliation);

    teamContainer.appendChild(memberDiv);
  });
});
</script>
</div>
</div>

  </main><aside class="col-md-3"><div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="https://opensource.org/licenses/mit-license.php" class="external-link">MIT</a></li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing HaploTraitR</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Alsamman Alsamman <br><small class="roles"> Author, maintainer </small>  </li>
<li><a href="authors.html">More about authors...</a></li>
</ul>
</div>



  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Alsamman Alsamman.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>





  </body>
</html>
