<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Package index â€¢ HaploTraitR</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Package index"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">HaploTraitR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-index">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Package index</h1>
    </div>

    <div class="section level2">
      <h2 id="all-functions">All functions<a class="anchor" aria-label="anchor" href="#all-functions"></a></h2>




    </div><div class="section level2">




      <dl><dt>

          <code><a href="boxplot_genotype_phenotype.html">boxplot_genotype_phenotype()</a></code>

        </dt>
        <dd>Generate boxplots for genotype-phenotype data</dd>
      </dl><dl><dt>

          <code><a href="combine_plots.html">combine_plots()</a></code>

        </dt>
        <dd>Function to combine multiple ggplot objects into a single gtable object</dd>
      </dl><dl><dt>

          <code><a href="computeLDclusterByChr.html">computeLDclusterByChr()</a></code>

        </dt>
        <dd>Compute LD matrix for a cluster</dd>
      </dl><dl><dt>

          <code><a href="computeLDclusters.html">computeLDclusters()</a></code>

        </dt>
        <dd>Compute LD matrix for clusters</dd>
      </dl><dl><dt>

          <code><a href="convertGenoBi2Numeric.html">convertGenoBi2Numeric()</a></code>

        </dt>
        <dd>Convert a habmap object to a numeric matrix</dd>
      </dl><dl><dt>

          <code><a href="convertLDclusters2Haps.html">convertLDclusters2Haps()</a></code>

        </dt>
        <dd>Convert LD clusters to haplotypes and calculate the frequency of each haplotype combination</dd>
      </dl><dl><dt>

          <code><a href="create_dirs.html">create_dirs()</a></code>

        </dt>
        <dd>Create directories for haplotype combination boxplots</dd>
      </dl><dl><dt>

          <code><a href="create_ld_plot.html">create_ld_plot()</a></code>

        </dt>
        <dd>Function to create LD plot and integrate with SNP positions</dd>
      </dl><dl><dt>

          <code><a href="create_unique_result_folder.html">create_unique_result_folder()</a></code>

        </dt>
        <dd>Create a unique result folder for HaplotraitR</dd>
      </dl><dl><dt>

          <code><a href="diamond_annots.html">diamond_annots()</a></code>

        </dt>
        <dd>Get diamond ggplot layer.</dd>
      </dl><dl><dt>

          <code><a href="evaluate_test_group0.html">evaluate_test_group0()</a></code>

        </dt>
        <dd>Evaluate haplotype combinations versus group 0 in each trait</dd>
      </dl><dl><dt>

          <code><a href="extract_hapmap.html">extract_hapmap()</a></code>

        </dt>
        <dd>Extract hapmap data for significant SNPs</dd>
      </dl><dl><dt>

          <code><a href="find_nearby_snps.html">find_nearby_snps()</a></code>

        </dt>
        <dd>Find nearby SNPs to a given SNP list assuming they are in the same chromosome</dd>
      </dl><dl><dt>

          <code><a href="generateHapCombBoxPlots.html">generateHapCombBoxPlots()</a></code>

        </dt>
        <dd>Generate and save haplotype combination boxplots</dd>
      </dl><dl><dt>

          <code><a href="getDistClusters.html">getDistClusters()</a></code>

        </dt>
        <dd>Get haplotype clusters from GWAS and hapmap data</dd>
      </dl><dl><dt>

          <code><a href="getDistClustersByChr.html">getDistClustersByChr()</a></code>

        </dt>
        <dd>Get Distance clusters from GWAS and hapmap data By chromosome</dd>
      </dl><dl><dt>

          <code><a href="getHapCombSamples.html">getHapCombSamples()</a></code>

        </dt>
        <dd>Get samples for a haplotype combination</dd>
      </dl><dl><dt>

          <code><a href="getLDclusters.html">getLDclusters()</a></code>

        </dt>
        <dd>Cluster SNPs based on LD matrices</dd>
      </dl><dl><dt>

          <code><a href="getSNPcombTables.html">getSNPcombTables()</a></code>

        </dt>
        <dd>Get SNP combinations sample tables</dd>
      </dl><dl><dt>

          <code><a href="get_config.html">get_config()</a></code>

        </dt>
        <dd>Get a configuration parameter</dd>
      </dl><dl><dt>

          <code><a href="get_pheno_geno.html">get_pheno_geno()</a></code>

        </dt>
        <dd>Get the phenotype and genotype data</dd>
      </dl><dl><dt>

          <code><a href="ggplot_ld.html">ggplot_ld()</a></code>

        </dt>
        <dd>Ggplot linkage disequilibrium</dd>
      </dl><dl><dt>

          <code><a href="merge_with_list_name.html">merge_with_list_name()</a></code>

        </dt>
        <dd>Merge a list of data frames with a list name</dd>
      </dl><dl><dt>

          <code><a href="plotCombMatrix.html">plotCombMatrix()</a></code>

        </dt>
        <dd>Plot Haplotype Combinations Matrix</dd>
      </dl><dl><dt>

          <code><a href="plotHapCombBoxPlot.html">plotHapCombBoxPlot()</a></code>

        </dt>
        <dd>Plotting Haplotype Combination Boxplot for specific SNP</dd>
      </dl><dl><dt>

          <code><a href="plotLDCombMatrix.html">plotLDCombMatrix()</a></code>

        </dt>
        <dd>Plot LD and Haplotype with Combination Matrix</dd>
      </dl><dl><dt>

          <code><a href="plotLDForClusters.html">plotLDForClusters()</a></code>

        </dt>
        <dd>Plot LD heatmap from LDs Info table</dd>
      </dl><dl><dt>

          <code><a href="plotLDheatmap.html">plotLDheatmap()</a></code>

        </dt>
        <dd>Plotting LD Heatmap for a Specific Haplotype</dd>
      </dl><dl><dt>

          <code><a href="plot_snp_positions.html">plot_snp_positions()</a></code>

        </dt>
        <dd>Function to plot SNP positions</dd>
      </dl><dl><dt>

          <code><a href="readGWAS.html">readGWAS()</a></code>

        </dt>
        <dd>Read GWAS results file with specified columns and FDR calculation</dd>
      </dl><dl><dt>

          <code><a href="readHapmap.html">readHapmap()</a></code>

        </dt>
        <dd>Read hapmap file into R</dd>
      </dl><dl><dt>

          <code><a href="readVCF.html">readVCF()</a></code>

        </dt>
        <dd>Read a VCF file</dd>
      </dl><dl><dt>

          <code><a href="saveTTestResultsToFile.html">saveTTestResultsToFile()</a></code>

        </dt>
        <dd>Convert t-test results to a data frame and save to a file</dd>
      </dl><dl><dt>

          <code><a href="set_config.html">set_config()</a></code>

        </dt>
        <dd>Set configuration parameters for HaploTraitR</dd>
      </dl><dl><dt>

          <code><a href="testSNPcombs.html">testSNPcombs()</a></code>

        </dt>
        <dd>Calculate the p-value for a SNP combinations</dd>
      </dl></div>
  </main></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Alsamman Alsamman.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer></div>





  </body></html>

