<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Package index â€¢ HaploTraitR</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Package index"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">


    <div class="container template-reference-index">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">HaploTraitR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/AlsammanAlsamman/HaploTraitR" class="external-link">GitHub</a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->



      </header><div class="row">
  <div class="contents col-md-9">
    <div class="page-header">
      <h1>Reference</h1>
    </div>

    <table class="ref-index"><colgroup><col class="alias"><col class="title"></colgroup><tbody><tr><th colspan="2">
          <h2 id="all-functions">All functions <a href="#all-functions" class="anchor" aria-hidden="true"></a></h2>
          <p class="section-desc"></p>
        </th>
      </tr></tbody><tbody><tr><td>
          <p><code><a href="boxplot_genotype_phenotype.html">boxplot_genotype_phenotype()</a></code> </p>
        </td>
        <td><p>Generate boxplots for genotype-phenotype data</p></td>
      </tr><tr><td>
          <p><code><a href="combine_plots.html">combine_plots()</a></code> </p>
        </td>
        <td><p>Function to combine multiple ggplot objects into a single gtable object</p></td>
      </tr><tr><td>
          <p><code><a href="computeLDclusterByChr.html">computeLDclusterByChr()</a></code> </p>
        </td>
        <td><p>Compute LD matrix for a cluster</p></td>
      </tr><tr><td>
          <p><code><a href="computeLDclusters.html">computeLDclusters()</a></code> </p>
        </td>
        <td><p>Compute LD matrix for clusters</p></td>
      </tr><tr><td>
          <p><code><a href="convertGenoBi2Numeric.html">convertGenoBi2Numeric()</a></code> </p>
        </td>
        <td><p>Convert a habmap object to a numeric matrix</p></td>
      </tr><tr><td>
          <p><code><a href="convertLDclusters2Haps.html">convertLDclusters2Haps()</a></code> </p>
        </td>
        <td><p>Convert LD clusters to haplotypes and calculate the frequency of each haplotype combination</p></td>
      </tr><tr><td>
          <p><code><a href="create_dirs.html">create_dirs()</a></code> </p>
        </td>
        <td><p>Create directories for haplotype combination boxplots</p></td>
      </tr><tr><td>
          <p><code><a href="create_ld_plot.html">create_ld_plot()</a></code> </p>
        </td>
        <td><p>Function to create LD plot and integrate with SNP positions</p></td>
      </tr><tr><td>
          <p><code><a href="create_unique_result_folder.html">create_unique_result_folder()</a></code> </p>
        </td>
        <td><p>Create a unique result folder for HaplotraitR</p></td>
      </tr><tr><td>
          <p><code><a href="diamond_annots.html">diamond_annots()</a></code> </p>
        </td>
        <td><p>Get diamond ggplot layer.</p></td>
      </tr><tr><td>
          <p><code><a href="evaluate_test_group0.html">evaluate_test_group0()</a></code> </p>
        </td>
        <td><p>Evaluate haplotype combinations versus group 0 in each trait</p></td>
      </tr><tr><td>
          <p><code><a href="extract_hapmap.html">extract_hapmap()</a></code> </p>
        </td>
        <td><p>Extract hapmap data for significant SNPs</p></td>
      </tr><tr><td>
          <p><code><a href="find_nearby_snps.html">find_nearby_snps()</a></code> </p>
        </td>
        <td><p>Find nearby SNPs to a given SNP list assuming they are in the same chromosome</p></td>
      </tr><tr><td>
          <p><code><a href="generateHapCombBoxPlots.html">generateHapCombBoxPlots()</a></code> </p>
        </td>
        <td><p>Generate and save haplotype combination boxplots</p></td>
      </tr><tr><td>
          <p><code><a href="getDistClusters.html">getDistClusters()</a></code> </p>
        </td>
        <td><p>Get haplotype clusters from GWAS and hapmap data</p></td>
      </tr><tr><td>
          <p><code><a href="getDistClustersByChr.html">getDistClustersByChr()</a></code> </p>
        </td>
        <td><p>Get Distance clusters from GWAS and hapmap data By chromosome</p></td>
      </tr><tr><td>
          <p><code><a href="getHapCombSamples.html">getHapCombSamples()</a></code> </p>
        </td>
        <td><p>Get samples for a haplotype combination</p></td>
      </tr><tr><td>
          <p><code><a href="getLDclusters.html">getLDclusters()</a></code> </p>
        </td>
        <td><p>Cluster SNPs based on LD matrices</p></td>
      </tr><tr><td>
          <p><code><a href="getSNPcombTables.html">getSNPcombTables()</a></code> </p>
        </td>
        <td><p>Get SNP combinations sample tables</p></td>
      </tr><tr><td>
          <p><code><a href="get_config.html">get_config()</a></code> </p>
        </td>
        <td><p>Get a configuration parameter</p></td>
      </tr><tr><td>
          <p><code><a href="get_pheno_geno.html">get_pheno_geno()</a></code> </p>
        </td>
        <td><p>Get the phenotype and genotype data</p></td>
      </tr><tr><td>
          <p><code><a href="ggplot_ld.html">ggplot_ld()</a></code> </p>
        </td>
        <td><p>Ggplot linkage disequilibrium</p></td>
      </tr><tr><td>
          <p><code><a href="merge_with_list_name.html">merge_with_list_name()</a></code> </p>
        </td>
        <td><p>Merge a list of data frames with a list name</p></td>
      </tr><tr><td>
          <p><code><a href="plotCombMatrix.html">plotCombMatrix()</a></code> </p>
        </td>
        <td><p>Plot Haplotype Combinations Matrix</p></td>
      </tr><tr><td>
          <p><code><a href="plotHapCombBoxPlot.html">plotHapCombBoxPlot()</a></code> </p>
        </td>
        <td><p>Plotting Haplotype Combination Boxplot for specific SNP</p></td>
      </tr><tr><td>
          <p><code><a href="plotLDCombMatrix.html">plotLDCombMatrix()</a></code> </p>
        </td>
        <td><p>Plot LD and Haplotype with Combination Matrix</p></td>
      </tr><tr><td>
          <p><code><a href="plotLDForClusters.html">plotLDForClusters()</a></code> </p>
        </td>
        <td><p>Plot LD heatmap from LDs Info table</p></td>
      </tr><tr><td>
          <p><code><a href="plotLDheatmap.html">plotLDheatmap()</a></code> </p>
        </td>
        <td><p>Plotting LD Heatmap for a Specific Haplotype</p></td>
      </tr><tr><td>
          <p><code><a href="plot_snp_positions.html">plot_snp_positions()</a></code> </p>
        </td>
        <td><p>Function to plot SNP positions</p></td>
      </tr><tr><td>
          <p><code><a href="readGWAS.html">readGWAS()</a></code> </p>
        </td>
        <td><p>Read GWAS results file with specified columns and FDR calculation</p></td>
      </tr><tr><td>
          <p><code><a href="readHapmap.html">readHapmap()</a></code> </p>
        </td>
        <td><p>Read hapmap file into R</p></td>
      </tr><tr><td>
          <p><code><a href="readVCF.html">readVCF()</a></code> </p>
        </td>
        <td><p>Read a VCF file</p></td>
      </tr><tr><td>
          <p><code><a href="saveTTestResultsToFile.html">saveTTestResultsToFile()</a></code> </p>
        </td>
        <td><p>Convert t-test results to a data frame and save to a file</p></td>
      </tr><tr><td>
          <p><code><a href="set_config.html">set_config()</a></code> </p>
        </td>
        <td><p>Set configuration parameters for HaploTraitR</p></td>
      </tr><tr><td>
          <p><code><a href="testSNPcombs.html">testSNPcombs()</a></code> </p>
        </td>
        <td><p>Calculate the p-value for a SNP combinations</p></td>
      </tr></tbody></table></div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by The package maintainer.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

      </footer></div>






  </body></html>

